bip-0360.mediawiki

```diff
@@ -0,0 +1,596 @@
+
+== Specification ==
+
+We define the signature scheme and transaction structure as follows.
+
+=== Descriptor Format ===
+
+To integrate P2QRH into existing wallet software and scripts, we introduce a new output descriptor function
+<code>qrh()</code>. This function represents a P2QRH output, similar to how <code>wpkh()</code> and <code>tr()</code>
+are used for P2WPKH and P2TR outputs, respectively.
+
+The <code>qrh()</code> function takes the HASH256 of the concatenated HASH256 of the quantum-resistant public keys as
+its argument. For example:
+
+  qrh(HASH256(HASH256(pubkey1) <nowiki>||</nowiki> HASH256(pubkey2) <nowiki>||</nowiki> ...))
+
+This function allows wallets to manage P2QRH addresses and outputs while accommodating multiple public keys of varying
+lengths, such as in multisig schemes, while keeping the public keys hidden until the time of spending.
+
+=== Address Format ===
+
+P2QRH uses SegWit version 3 outputs, resulting in addresses that start with <code>bc1r</code>, following
+[https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki#bech32 BIP-173]. Bech32 encoding maps version 3 to the
+prefix <code>r</code>.
+
+Example P2QRH address:
+
+<code>bc1r...</code> (32-byte Bech32m-encoded HASH256 of the HASH256 of the public keys)
+
+=== ScriptPubKey ===
+
+The <code>scriptPubKey</code> for a P2QRH output is:
+
+  OP_PUSHNUM_3 OP_PUSHBYTES_32 <nowiki><hash></nowiki>
+
+Where:
+
+* <code>OP_PUSHNUM_3</code> (<code>0x03</code>) indicates SegWit version 3.
+* <nowiki><hash></nowiki> is the 32-byte HASH256 of the concatenated HASH256 of each public key.
+
+==== Hash Computation ====
+
+The hash is computed as a merkle tree of public key hashes:
+
+1. For each public key, compute its HASH256
+2. Pair the hashes and compute HASH256 of their concatenation
+3. Continue pairing and hashing until reaching a single root hash
+4. The final hash is the merkle root
+
+For example with 4 public keys:
+
+  h1 = HASH256(pubkey1)
+  h2 = HASH256(pubkey2)
+  h3 = HASH256(pubkey3)
+  h4 = HASH256(pubkey4)
+
+  h12 = HASH256(h1 <nowiki>||</nowiki> h2)
+  h34 = HASH256(h3 <nowiki>||</nowiki> h4)
+
+  root = HASH256(h12 <nowiki>||</nowiki> h34)
+
+When spending, if a public key hash is provided in the attestation with an empty signature, that hash will be used
+directly in the merkle tree computation rather than hashing the full public key. This allows excluding unused public
+keys from the transaction while still proving they were part of the original commitment.
+
+This merkle tree construction creates an efficient cryptographic commitment to multiple public keys while enabling
+selective disclosure.
+
+This allows for inclusion of a Taproot MAST merkle root in the attestation, which makes P2QRH a quantum-resistant
+version of Taproot.
+
+=== Transaction Serialization ===
+
+Following BIP-141, the transaction serialization is modified to include a new attestation field after the witness field:
```

Comment:
This is a bit inaccurate: you can't just modify the existing serialization format because existing nodes still need to receive transactions in that format. Instead you're creating a new format and keeping both (actually 3 now).

By the way, will you not also need to introduce e.g. a `qtxid`, similar to `wtxid`?
```suggestion
Following BIP-141, a new transaction serialization format is introduced to include an attestation field after the witness field:
```